File: quickbytes_cloud/functions/src/models/article.ts
export enum Relevancy {
  all = "all",
  major = "major",
}

export interface Article {
  id?: string;
  title: string;
  content: string;
  published_on: Date;
  category_list: [];
  relevancy: Relevancy;
  source_url: string;
  image: string;
}

File: quickbytes_app/android/app/src/main/kotlin/com/quickbytes/quickbytes_app/Models/Article.kt
import kotlinx.serialization.*

@Serializable
data class Article (
    val id: String,
    val title: String,
    val content: String,
    val image: String,

    @SerialName("published_on")
    val publishedOn: String,

    @SerialName("category_list")
    val categoryList: List<String>,

    val relevancy: String,

    @SerialName("source_url")
    val sourceURL: String
)
